framework/src/outputs/Checkpoint.C

```diff
@@ -75,60 +87,80 @@ Checkpoint::directory() const
 void
 Checkpoint::output(const ExecFlagType & /*type*/)
 {
-  // Create the output directory
-  std::string cp_dir = directory();
-  Utility::mkdir(cp_dir.c_str());
+  // Check if we should write the autosave checkpoint. The only time _should_output will
+  // be false is if this Checkpoint object is created through AutoCheckpointAction, and if it
+  // is false, then we check if the signal handler has set the flag for us to write it out.
 
-  // Create the output filename
-  std::string current_file = filename();
+  if (!_should_output)
+  {
+    comm().max(Moose::autosave_flag);
```

Comment:
OK - so no more parallel sync issues. However, as I mentioned this can be surprisingly expensive in a sim with a low number of DOFs. We may want to do several things here:
- We can avoid the communication altogether if `_should_output` is already true (outside of the global variable). This would be the normal intentional addition of the Checkpoint object. 
- Actually the above statement is only partially true. If we can add multiple Checkpoints (should be possible, even if slightly misguided), we should only ever need to do this communication in the "autosave" version depending on how you handle that situation ... OR we don't add a separate object if it's already turned on and make it do double duty. We need to address this case and handle it one way or the other. You might be writing checkpoints twice if it's explicitly turned on and also turned on via signals. 
- Finally, we may want to address the Multiapp case now too. Talk to me about this more in the office, but in a nutshell, we don't want to add the autosave option to Multiapps. We should check that in the Action (see `MooseApp::isUltimateMaster`).